rainbows.rb

```diff
@@ -1,29 +1,29 @@
-red = [255,0,0]
-green = [0,255,0]
-blue = [0,0,255]
+require "colorize"
 
-color_lookup = {
+#1
+# Let's create a rainbow of colors as an array of hashes,
+# where each hash contains the name of the color (key) and
+# the RGB values (value).
+
+rainbow_color= {
   [255,0,0] => "red",
-  [255,125,0] => "orange",
+  [255,127,0] => "orange",
   [255,255,0] => "yellow",
-  [125,255,0] => "spring green",
   [0,255,0] => "green",
-  [0,255,125] => "turquoise",
-  [0,255,255] => "cyan",
-  [0,125,255] => "ocean",
   [0,0,255] => "blue",
-  [125,0,255] => "violet",
-  [255,0,255] => "magenta",
-  [255,0,125] => "raspberry"
+  [75,0,130] => "indigo",
+  [143, 0, 255] => "violet"
 }
 
-def mix_colors(color1,color2)
-  mixed_color = []
-  3.times do |i|
-    mixed_color[i] = color1[i] + color2[i]
-  end
-  return mixed_color
+# create a rainbow of colors as an array of hashes rainbow[0] = :red => {r: 255, g: 0, b: 0}
+rainbow = rainbow_color.map do |rgb_array, color|
+  {color => {r: rgb_array[0], g: rgb_array[1], b: rgb_array[2]}}
 end
 
-new_color = mix_colors(red,green)
-puts "#{color_lookup[red]} #{red} + #{color_lookup[green]} #{green} = #{color_lookup[new_color]} #{new_color}"
+#2 Use your rainbow array to print out each color name in sequence.
+# (optional) Print the color name in the color it represents using the colorize gem
+
+rainbow.each do |rainbow_color|
+  color = rainbow_color.keys.first
+  puts "#{color}".colorize(color.to_sym)
+end
```

Comment:
It doesn't capture all the colors. I'm missing indigo and violet.

On Thu, Mar 31, 2016 at 10:14 AM, Crystal <notifications@github.com> wrote:

> In rainbows.rb
> <https://github.com/Ada-C5/rainbows/pull/20#discussion_r58091427>:
>
> >  end
> >
> > -new_color = mix_colors(red,green)
> > -puts "#{color_lookup[red]} #{red} + #{color_lookup[green]} #{green} = #{color_lookup[new_color]} #{new_color}"
> > +#2 Use your rainbow array to print out each color name in sequence.
> > +# (optional) Print the color name in the color it represents using the colorize gem
> > +
> > +rainbow.each do |rainbow_color|
> > +  color = rainbow_color.keys.first
> > +  puts "#{color}".colorize(color.to_sym)
> > +end
>
> Looks good! Does this do all the colors with colorize? Or miss a few?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/Ada-C5/rainbows/pull/20/files/4f60b5cb497363fc805ff1307db99e471a7a33e8#r58091427>
>